File: server/BillPollAPI/WebApplication1/API/Poll.cs
namespace WebApplication1.API
{
    public class Poll
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public ICollection<Option> Options { get; set; } = new List<Option>();

        public Poll(Models.Poll model)
        {
            this.Id = model.Id;
            this.Name = model.Name;
            foreach (Models.Option option in model.Options)
            {
                this.Options.Add(new Option(option));
            }

        }

        public Poll(int id, string name, ICollection<Option> options)
        {
            Id = id;
            Name = name;
            Options = options;
        }
        public Poll() { }
    }

    public class PollCreate
    {
        public string Name { get; set; }
        public ICollection<OptionCreate> Options { get; set; } = new List<OptionCreate>();

        public PollCreate(string name, ICollection<OptionCreate> options)
        {
            Name = name;
            Options = options;
        }
        public PollCreate() { }
    }
}

File: billpoll/src/remote/poll.ts
import { PollCreate } from "../interfaces/interfaces";

const SERVER_URL = "http://localhost:5295";

export const GetPolls = async (): Promise<any> => {
  const res = await fetch(SERVER_URL+"/poll", { cache: "no-store" });
  if (!res.ok) {
            throw new Error(
              `Could not get response: ${res.status} ${res.statusText}`
            );
  }

  return res.json()
};


export const CreatePoll = async(name: string, options: string[]) : Promise<any> => {
    let toCreate: PollCreate = { name: name, options: [] };
    options.forEach((option) => {
        toCreate.options.push({ name: option });
    });

    const res = await fetch(SERVER_URL + "/poll", {
      method: "POST",
      body: JSON.stringify(toCreate),
      headers: {
        "Content-Type": "application/json",
      },
    });

    if (!res.ok) {
        throw new Error(
          `Failed to create poll: ${res.status} ${res.statusText}`
        );
    }

     return res.json();
}